cobra/io/sbml.py

```diff
@@ -779,10 +779,14 @@ def process_association(ass):
 
     cobra_model.add_groups(groups)
 
-    # run the complete parsing to get all warnings and errors before
-    # raising errors
+    # general hint for missing flux bounds
     if missing_bounds and not set_missing_bounds:
-        raise CobraSBMLError("Missing flux bounds on reactions.")
+        LOGGER.error("Missing flux bounds on reactions. As best practise and "
```

Comment:
Shouldn't this be a warning? Also, when the individual missing bounds are raised? I mean for example `ERROR:cobra.io.sbml:Missing lower flux bound for reaction: <Reaction R00291>` -> `WARNING:cobra.io.sbml:Missing lower flux bound for reaction: <Reaction R00291>`. I completely agree that bounds should always be set but FBC allows undefined bounds, right? Also, I checked that currently a model without bounds is parsed and returned with default bounds 0 and 1000 even if `set_missing_bounds=False`. What is the anticipated behavior when `set_missing_bounds=False`?

Reply:
This is exactly the problem here. Without defined bounds the LP problem is not completely defined and every software is doing what they want. Depending on where you look you will find:
- i) setting no bounds, i.e. adding no constraints to the LP (or using [-INF, INF] which is equivalent); This is the correct solution.
- ii) setting [-INF, INF], [0, INF], [-INF, 0] bounds (using reversible information or not); There could be a long discussion here about what is correct or not. Both sides of the argument are somehow right :/
- iii) setting [-c, c], [0, c], [-c, 0] bounds (using reversible information or not) with c being an arbitrary value (in case of cobrapy being c=1000). This is definitely wrong.

As a consequence a model with missing bounds is not a completely specified LP (it would be if people would do the correct thing of not doing anything, i.e. just doing i) instead of trying to be clever and doing things like ii) or iii). As a consequence different tools will give different results.
There was a long discussion here that cobrapy should only load valid (and uniquely defined LP problems), because in the case of missing bounds this is unfortunately not the case I decided to give an error. But apparently many models don't have bounds on all reactions (?!)
I did not realize that cobrapy is doing iii) when creating a reaction, which is incorrect behavior.

**Changes**
- yes, bounds are not required. I will change to warnings.
- expected behavior for `set_missing_bounds=False``is to have [-INF, INF] bounds because there are no bounds specified, injecting arbitrary bounds in the LP is incorrect. So I have to set the bounds explicitly to [-INF, INF] for missing bounds on import because on reaction creation arbitrary bounds are set.

Reply:
Thinking a minute more about it: Setting arbitrary values (i.e. the values in cobra.configuration) could work and is probably what people in the COBRA community expect. But this has to be clearly communicated in the warning. So missing bounds would just filled in with the cobra.configuration bounds. 

When I load my models I will just use the correct `[-INF, INF]` in cobra.configuration. 
This is basically #815 and #816. Personally, using default bounds feels very wrong and just screams irreproducibility of results, because things could depend on which settings one has in cobra.configuration for the default bounds.

Reply:
Don't know, I had expected that `set_missing_bounds=False` would raise an exception and let the import fail. Just to clarify, what you're suggesting is?

* (i) `set_missing_bounds=False`: set bounds to [-INF, INF] (where INF == `float("inf")`) raising warnings that this is happening. No reversibility information is used otherwise. 
* (iii)`set_missing_bounds=True`: set bounds to whatever is defined cobra.configuration.

I think this is a perfectly fine solution. In memote we would stick to `set_missing_bounds=False` then I think.